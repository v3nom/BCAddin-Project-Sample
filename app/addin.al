controladdin DemoAddin
{
    Scripts = 'Scripts/bundle.js';
    StartupScript = 'Scripts/start.js';

    RequestedWidth = 800;
    RequestedHeight = 450;
    MinimumWidth = 800;
    MinimumHeight = 450;
    MaximumWidth = 800;
    MaximumHeight = 450;
    HorizontalShrink = true;
    VerticalShrink = true;
    HorizontalStretch = true;
    VerticalStretch = true;

    event ControlReady()

    event UserInput(value: Text)

    procedure Message(value: Text)
}