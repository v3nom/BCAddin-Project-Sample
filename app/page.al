page 50100 DemoAddinpage
{
    PageType = Card;
    ApplicationArea = All;
    UsageCategory = Lists;
    Editable = false;

    layout
    {
        area(Content)
        {
            usercontrol(addin; DemoAddin)
            {
                ApplicationArea = All;
                trigger ControlReady()
                begin

                end;

                // Value is received from the browser addin
                trigger UserInput(value: Text)
                var
                    ok: Boolean;
                begin
                    ok := Dialog.Confirm('AL: Do you accept new value?');
                    if ok then begin
                        CurrPage.addin.Message(Format(value, 0, 'Message from AL: <Text>, has been accepted!'));
                    end else begin
                        Dialog.Message('AL: User rejected the value from browser addin');
                    end;
                end;
            }
        }
    }
}