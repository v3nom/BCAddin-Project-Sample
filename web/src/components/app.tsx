import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { Button } from 'office-ui-fabric-react/lib/Button';
import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

// Sends user value back to AL
function sendInputToAL(value: string) {
    Microsoft.Dynamics.NAV.InvokeExtensibilityMethod("UserInput", [value]);
}

// Main React component
export const AppComponent = (props: { message?: string }) => {
    const [value, setValue] = React.useState(props.message);
    return <Fabric>
        {props.message && <MessageBar messageBarType={MessageBarType.success}>{props.message}</MessageBar>}
        <TextField value={value} autoComplete='off' onChange={(_, newValue) => setValue(newValue)} placeholder="Enter your value" />
        <Button onClick={() => sendInputToAL(value)}>Submit</Button>
    </Fabric>;
};
