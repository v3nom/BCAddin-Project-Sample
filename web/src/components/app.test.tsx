import * as React from "react";
import { AppComponent } from "./app";
import * as Renderer from "react-test-renderer";


describe("App Component test suite", () => {
    test("Render without success message", () => {
        const component = Renderer.create(<AppComponent></AppComponent>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Render with success message", () => {
        const component = Renderer.create(<AppComponent message="Value accepted"></AppComponent>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});