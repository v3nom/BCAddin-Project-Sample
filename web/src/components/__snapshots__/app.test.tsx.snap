// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component test suite Render with success message 1`] = `
<div
  className="ms-Fabric root-40"
>
  <div
    style={
      Object {
        "background": "#ffffff",
      }
    }
  >
    <div
      className="ms-MessageBar ms-MessageBar--success ms-MessageBar-multiline root-59"
    >
      <div
        className="ms-MessageBar-content content-60"
      >
        <div
          aria-hidden={true}
          className="ms-MessageBar-icon iconContainer-61"
        >
          <i
            aria-hidden={true}
            className="icon-70"
            data-icon-name="Completed"
          />
        </div>
        <div
          aria-live="polite"
          className="ms-MessageBar-text text-63"
          id="MessageBar6"
          role="status"
        >
          <span
            className="ms-MessageBar-innerText innerText-64"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ms-TextField root-42"
  >
    <div
      className="ms-TextField-wrapper wrapper-43"
    >
      <div
        className="ms-TextField-fieldGroup fieldGroup-44"
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          className="ms-TextField-field field-45"
          id="TextField7"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder="Enter your value"
          type="text"
          value="Value accepted"
        />
      </div>
    </div>
  </div>
  <button
    className="ms-Button ms-Button--default root-51"
    data-is-focusable={true}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="ms-Button-flexContainer flexContainer-52"
      data-automationid="splitbuttonprimary"
    >
      <span
        className="ms-Button-textContainer textContainer-53"
      >
        <span
          className="ms-Button-label label-55"
          id="id__10"
        >
          Submit
        </span>
      </span>
    </span>
  </button>
</div>
`;

exports[`App Component test suite Render without success message 1`] = `
<div
  className="ms-Fabric root-40"
>
  <div
    className="ms-TextField root-42"
  >
    <div
      className="ms-TextField-wrapper wrapper-43"
    >
      <div
        className="ms-TextField-fieldGroup fieldGroup-44"
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          className="ms-TextField-field field-45"
          id="TextField0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          placeholder="Enter your value"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <button
    className="ms-Button ms-Button--default root-51"
    data-is-focusable={true}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="ms-Button-flexContainer flexContainer-52"
      data-automationid="splitbuttonprimary"
    >
      <span
        className="ms-Button-textContainer textContainer-53"
      >
        <span
          className="ms-Button-label label-55"
          id="id__3"
        >
          Submit
        </span>
      </span>
    </span>
  </button>
</div>
`;
